core/iwasm/common/wasm_application.c

```diff
@@ -220,6 +221,12 @@ wasm_application_execute_main(WASMModuleInstanceCommon *module_inst, int32 argc,
     wasm_runtime_dump_perf_profiling(module_inst);
 #endif
 
+#if WASM_ENABLE_DUMP_CALL_STACK != 0
+    exec_env = wasm_runtime_get_exec_env_singleton(module_inst);
+    if (!ret && exec_env)
+        wasm_runtime_dump_call_stack(exec_env);
```

Comment:
isn't it simpler to do this only when the following `wasm_runtime_get_exception` returns true?